content/docs/reference/buildpack-api.md

```diff
@@ -203,6 +203,23 @@ A buildpack can `require` or `provide` multiple dependencies, and even multiple
 The lifecycle uses the `Build Plan` as one element in deciding whether or not a particular list of buildpacks is appropriate, by seeing whether all dependencies required can be provided by that list. See the [spec](https://github.com/buildpacks/spec/blob/master/buildpack.md#phase-1-detection) for particulars on how ordering buildpacks can adjust detection results.
 
 ### Example
+Let's walk through some possible cases a `node-engine` buildpack may consider:
+
+1.  Nothing in the app explicitly calls out that it is needed
+2.  It is explicitly referred to in some configuration file
+3.  Two different configuration files request different versions of `node`
+
+We will also consider what a `NPM` buildpack may do.
```

Comment:
It might be helpful (speaking as someone who doesn't know node) to call out what is the responsibility of the node-engine buildpack vs. the NPM buildpack.